src/components/molecules/Tab/Tab.test.tsx

```diff
@@ -0,0 +1,58 @@
+import "@testing-library/jest-dom";
+import { render, fireEvent } from "@testing-library/react";
+import { describe, test } from "vitest";
+
+import { TabArea } from "./TabArea/TabArea";
+import { TabNavigation } from "./TabNavigation/TabNavigation";
+import { TabPage } from "./TabPage/TabPage";
+
+const Tab = () => (
+  <TabArea>
+    <TabNavigation
+      tabs={[
+        { label: "page1", pageId: 1 },
+        { label: "page2", pageId: 2 },
+        { label: "page3", pageId: 3 },
+      ]}
+    />
+    <div>
+      <TabPage page={1}>Content 1</TabPage>
+      <TabPage page={2}>Content 2</TabPage>
+      <TabPage page={3}>Content 3</TabPage>
+    </div>
+  </TabArea>
+);
+
+describe("Tab", () => {
+  test("renders", async () => {
+    render(<Tab />);
+  });
+
+  test("Switches pageId on click", async () => {
+    const tab = render(<Tab />);
+
+    const page1Button = await tab.findByTestId("navigation-button__1");
+    const page2Button = await tab.findByTestId("navigation-button__2");
```

Comment:
When querying for accessible and semantic HTML elements I usually prefer to use `getByRole`. Since the buttons or tabs do have names, it should be quite easy to query for them using the role and the name. Here is a suggestion on how to write it using the data from the story for storybook:
```suggestion
    const page1Button = screen.getByRole("button", {name: "page1"})
    const page2Button = screen.getByRole("button", {name: "page2"})
```
(This currently doesn't work because the label for the tabs need to be different for it. And in real use case I think they would be.)